### PR TITLE
Ops 992 notification api

### PR DESCRIPTION
**Pull request by johndeange**
## What changed

Added initial Notification Center.

## Issue

https://app.zenhub.com/workspaces/opre-ops-62d197b4c468970013bcd2ec/issues/gh/hhs/opre-ops/992

## How to test

1. Log into the application and click the bell icon next to your email.
2. Add a Notification object to the DB
3. The notification should appear in the UI within 5 s

### CODE REVIEW

File: frontend/src/components/UI/Header/Header.jsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import AuthSection from "../../Auth/AuthSection";
 import { Menu } from "./Menu";
 import { Search } from "./Search";
 import logo from "./OPRE_Logo.png";
+import NotificationCenter from "../NotificationCenter/NotificationCenter";
```
Comment (fpigeonjr) [inline frontend/src/components/UI/Header/Header.jsx:5]: linter error for being unused?

File: frontend/src/components/UI/Header/Search.jsx [hunk 2]
```diff
@@ -1,18 +1,21 @@
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { solid } from "@fortawesome/fontawesome-svg-core/import.macro";
+import NotificationCenter from "../NotificationCenter/NotificationCenter";
```
Comment (fpigeonjr) [inline frontend/src/components/UI/Header/Search.jsx:3]: linter error

File: frontend/src/components/UI/Notification/Notification.jsx
Comment (fpigeonjr) [inline frontend/src/components/UI/Notification/Notification.jsx:1]: can we add an `index.js` in this file with the default export for Notifcation.jsx? This way we can import in other files like `import Notification from '../Notification` without the extra `Notification/Notification`?

File: frontend/src/components/UI/Modal/Modal.jsx [hunk 3]
```diff
@@ -70,7 +71,14 @@ export const Modal = ({
                                 >
                                     {heading}
                                 </h2>
-                                {description && (
+                                {description && description instanceof Array && (
```
Comment (fpigeonjr) [inline frontend/src/components/UI/Modal/Modal.jsx:74]: I am seeing this warning in terminal

Failed prop type: Invalid prop `description` of type `array` supplied to `Modal`, expected `string`.

Comment (fpigeonjr) [inline frontend/src/components/UI/Modal/Modal.jsx:74]: wonder if we need to check array length is more than one here?
`description.length > 0 && (
// render the thing
)`

File: openapi.yml [hunk 4]
```diff
@@ -879,6 +879,99 @@ paths:
         '500':
           description: Internal Server Error
 
+  /api/v1/notifications/:
```
Comment (fpigeonjr) [inline openapi.yml:882]: nice, glad we are keeping this up-to-date

**Review by fpigeonjr — CHANGES_REQUESTED** [review]
tested locally and in `development` and looks good. 
Found some minor linting issues and suggestions for Modal and new Notification component.

**Comment by jonnalley** [comment]
This is a great step forward

**Review by fpigeonjr — APPROVED** [review]
thanks @johndeange.

**Review by stevtek — APPROVED** [review]

**Pull request merged**